src/Select.js

```diff
@@ -901,6 +901,25 @@ class Select extends React.Component {
 	filterOptions (excludeOptions) {
 		var filterValue = this.state.inputValue;
 		var options = this.props.options || [];
+
+		var extraOptions = this.props.extraOptions || [];
+		var labelKey = this.props.labelKey;
+		if (extraOptions.length) {
+			const filterValueLower = filterValue.toLowerCase();
+			const optionsSet = new Set();
+			options.forEach((value) => {
+				if (typeof value === 'object') {
+					optionsSet.add(`${value[labelKey]}`.toLowerCase());
+				}
+			});
+			const filteredExtraOptions = extraOptions.filter((value) => {
+				if (typeof value !== 'object') return false;
+				const label = `${value[labelKey]}`.toLowerCase();
+				return !optionsSet.has(label) && label.indexOf(filterValueLower) >= 0;
+			});
+			options = options.concat(filteredExtraOptions);
```

Comment:
Does this end up adding the new options to the very end? I think in this case the new option they want is probably the one they want, can we have it at the top instead?
```
options = [...filteredExtraOptions, ...options];
```

I haven't actually tried it though, so they may get resorted from this array anyways.

Reply:
Changed the order, as the component doesn't sort the options